Add IBC Relayer guide (#777)
celestiaorg/docs: docs/developers/ibc-relayer.md

---
sidebar_label: IBC Relaying Guide
description: Learn how to establish IBC connections and relay packets.
---

# IBC Relaying Guide

Celestia uses [IBC](https://ibcprotocol.dev/)
(Inter-Blockchain Communication protocol) to enable cross-chain
transfer of tokens. To support this capability it relies on
relayers, processes that can be run by anyone which constantly
scan for outbound packets on one chain and submits these packets
alongside corresponding proofs on the destination chain. This
section describes how one can setup a relayer and create new connections between chains. There
are two standard implementations:

- [Hermes](https://hermes.informal.systems/) built in Rust
- [Go Relayer](https://pkg.go.dev/github.com/cosmos/relayer) built in Go

> The following guide explains how to establish IBC connections and relay packets between the **Celestias Blockspace Race** and **Dymension 35-C** networks by using the Hermes relayer.

Reviewer: ```suggestion
The following guide explains how to establish IBC connections and relay packets between **Celestia's Blockspace Race** and **Dymension's 35-C** networks by using the Hermes relayer.

_This guide was adapted from https://medium.com/qubelabs/ibc-connection-and-relaying-celestia-bsr-dymension-35-c-e4423699e856._
```

## Setup

> [Hermes](https://hermes.informal.systems/quick-start/) is an open-source Rust implementation of an Inter-Blockchain Communication protocol (IBC) relayer released as part of the ibc-relayer-cli crate. It includes a CLI for relaying packets between Cosmos SDK chains, as well as Prometheus metrics and a REST API.

Reviewer: ```suggestion
[Hermes](https://hermes.informal.systems/quick-start/) is an open-source Rust implementation of an Inter-Blockchain Communication protocol (IBC) relayer released as part of the `ibc-relayer-cli` crate. It includes a CLI for relaying packets between Cosmos SDK chains, as well as Prometheus metrics and a REST API.
```

### Install Rust

```bash
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
source "$HOME/.cargo/env"

sudo apt-get install pkg-config libssl-dev
sudo apt install librust-openssl-dev build-essential git
```

### Build Hermes

Create a new directory and clone the Hermes repository. For the latest available release see Hermes [GitHub](https://github.com/informalsystems/hermes/tags).

```bash
mkdir -p $HOME/hermes
git clone https://github.com/informalsystems/hermes
cd hermes
git checkout v1.5.0
cargo install ibc-relayer-cli --bin hermes --locked
```
Create a `keys` directory and copy the template `config.toml` to the `.hermes` directory.

```bash
 mkdir -p $HOME/.hermes
 mkdir -p $HOME/.hermes/keys
 cp config.toml $HOME/.hermes
 ```

 Before proceeding, verify that Hermes is installed correctly by running `hermes version`.

 ## Hermes Configuration

 After you have successfully installed Hermes and created the necessary folders, you now have to edit `config.toml` and add the appropriate configurations for the chains you want to relay between.

For this tutorial, we will be using the following chains:

> - Celestia Blockspace Race `blockspacerace-0`

Reviewer: ```suggestion
- Celestia's Blockspace Race `blockspacerace-0`
```

 https://docs.celestia.org/nodes/blockspace-race/

> - Dymension 35-C `35-C`

Reviewer: ```suggestion
- Dymension's 35-C `35-C`
```
